devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java

```diff
@@ -0,0 +1,134 @@
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.awaitility.Awaitility.await;
+
+public class FastJarFormatWorksTest extends QuarkusGradleTestBase {
+
+    private Future<?> quarkusDev;
+
+    @Test
+    public void testFastJarFormatWorks() throws IOException, URISyntaxException {
+
+        final File projectDir= getProjectDir("test-that-fast-jar-format-works-for-gradle");
+
+        GradleRunner.create()
+                .forwardOutput()
+                .withPluginClasspath()
+                .withArguments(arguments("package", "DskipTests","-Dquarkus.package.type=fast-jar"))
+                .withProjectDir(projectDir)
+                .build();
+
+        Path jar=projectDir.toPath().toAbsolutePath().resolve(Paths.get("target/acme-1.0-SNAPSHOT-runner.jar"));
+        Assertions.assertFalse(Files.exists(jar));
+
+        jar = projectDir.toPath().toAbsolutePath().resolve(Paths.get("target/quarkus-app/quarkus-run.jar"));
+        Assertions.assertTrue(Files.exists(jar));
+
+        File output = new File(projectDir, "target/output.log");
+        output.createNewFile();
+
+        Process process = doLaunch(jar, output);
+        try {
+            // Wait until server up
+            await()
+                    .pollDelay(1, TimeUnit.SECONDS)
+                    .atMost(1, TimeUnit.MINUTES).until(() -> DevModeTestUtils.getHttpResponse("/app/hello/package", 200));
+
+            String logs = FileUtils.readFileToString(output, "UTF-8");
+
+            assertThatOutputWorksCorrectly(logs);
+
+            // test that the application name and version are properly set
+            getHttpResponse();
+            assertResourceReadingFromClassPathWorksCorrectly("");
+        } finally {
+            process.destroy();
+        }
+
+    }
+
+    private Process doLaunch(Path jar, File output) throws IOException {
+        List<String> commands = new ArrayList<>();
+        commands.add(JavaBinFinder.findBin());
+        commands.add("-jar");
+        commands.add(jar.toString());
+        ProcessBuilder processBuilder = new ProcessBuilder(commands.toArray(new String[0]));
+        processBuilder.redirectOutput(output);
+        processBuilder.redirectError(output);
+        return processBuilder.start();
+    }
+
+    static void assertResourceReadingFromClassPathWorksCorrectly(String path) {
```

Comment:
This should probably also be replaced with the proper `getHttpResponse`

Reply:
Okayy, will do